### PR TITLE
Accept DEBUG environment variable in Dockerfile

### PR DESCRIPTION
**Pull request by eatonphil**
I was trying out the :debug image and noticed it wasn't generating stack traces. The scripts/install.sh was correct but the way the DEBUG=true flag was being passed in in Github CI wasn't being applied. You need to add this [ARG](https://docs.docker.com/engine/reference/builder/#arg) directive for Docker to pass along `--build-arg $ARG=$VALUE`s.

I made an attempt to test that the production image has a production TigerBeetle build and same for the debug image.

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
       - name: Build and push
-        id: docker_build
+        id: docker_build1
```
Comment (sentientwaffle) [inline .github/workflows/docker.yml]: Can we use more descriptive ids to differentiate these? e.g. `docker_build_release` and `docker_build_debug`.

Comment (eatonphil) [reply .github/workflows/docker.yml]: Good point, done

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -67,21 +67,28 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
       - name: Login to GHCR
-        if: github.event_name != 'pull_request'
+        #if: github.event_name != 'pull_request'
```
Comment (sentientwaffle) [inline .github/workflows/docker.yml]: Does this mean it will run on every commit?

Comment (eatonphil) [reply .github/workflows/docker.yml]: I was just doing this to test the logic. I need to remove before merging :)

Comment (eatonphil) [reply .github/workflows/docker.yml]: (it has been removed)

**Review by sentientwaffle — APPROVED** [review]

**Pull request merged**